### PR TITLE
Refactor WorkletGlobalScopeType to Suppress Clippy Warning

### PR DESCRIPTION
**Pull request by rexbrown21**
This PR suppresses the Clippy warning for the new function in WorkletGlobalScopeType by adding the #[allow(clippy::new_ret_no_self)] attribute to the module. This change helps prevent unnecessary linting errors while keeping the existing behavior intact. 

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

### CODE REVIEW

**Comment by simonwuelker** [comment]
You need to sign your commits, you can do that by passing the `-s` flag to `git commit`. See https://book.servo.org/contributing.html#pull-request-checklist for more info

File: components/devtools/actors/inspector/page_style.rs [hunk 1]
```diff
@@ -206,7 +206,8 @@ impl PageStyleActor {
                             is_system: false,
                             inherited: inherited.clone(),
                         })
-                    });
+                    })
+                    .collect();
```
Comment (sagudev) [inline components/devtools/actors/inspector/page_style.rs:210]: Which clippy error requested this? I think this causes another allocation.

**Comment by mukilan** [comment]
@rexbrown21 It seems like you've made the changes for the `new_ret_no_self` warning on top of your branch for the PR [#33743](https://github.com/servo/servo/pull/33743/files), so the current PR contains changes not related to the `new_ret_no_self` change.

I suggest you create a new branch based on main and cherry-pick the relevant commit (`git cherry-pick dcf4871a0`) to include only the changes related to  `new_ret_no_self`.

**Comment by mukilan** [comment]
Closing this PR in favor of #33831.

**Pull request closed without merging**